pkg/trace/pb/decoder_bytes_test.go

```diff
@@ -83,3 +84,22 @@ func TestDecodeInvalidUTF8Bytes(t *testing.T) {
 	}
 	assert.ElementsMatch(t, accept, got)
 }
+
+func TestDecodeNilStringInMap(t *testing.T) {
```

Comment:
This test could be added to `TestMetaMapDeserialization` and `TestMetricsMapDeserialization` in spans_gen_modifs_tests.go which are already validating all the custom modifications on top of the generated deserializer.